### ISSUE TITLE
[BUG] Gesture detection inconsistency with different `children` orders

### ISSUE THREAD

**Issue by daniloapr** [fleaflet/flutter_map#1459]
### What is the bug?

The gesture detection in a Marker is ignored when it is "below" a CircleLayer, even if the marker is not inside the circle.

This video shows it. I added a `InkWell` to the "arrow" markers, so it's possible to see the pointer changing, indicating the component is selectable, with the `CircleLayer` **before** the `MarkerLayer` in the list (i.e: bottom layer). If the position is changed, with the `CircleLayer` **after** the `MarkerLayer`, the `InkWell` in the Markers becomes undetectable.

https://user-images.githubusercontent.com/31525418/223156238-16d2f9c7-f9d5-4f08-aa1e-da11b1ea2c34.mov

Layers with marker gesture detections working:
```dart
    children: [
        PolylineLayer(
          polylines: coordinates.toLatLng,
        ),
        CircleLayer(
          circles: _buildCircles(),
        ),
        MarkerLayer(
          markers: _buildTracklogMarkers(coordinates),
        ),
    ]
```
Layers with marker gesture detections NOT working:
```dart
    children: [
        PolylineLayer(
          polylines: coordinates.toLatLng,
        ),
        MarkerLayer(
          markers: _buildTracklogMarkers(coordinates),
        ),
        CircleLayer(
          circles: _buildCircles(),
        ),
    ]
```

### What is the expected behaviour?

- The `Marker`s inside and below the `Circle`s should be detectable if the `Circle` doesn't consume the pointer.
- The `Marker`s inside and below the `Circle`s should not be detectable if the `Circle` consumes the pointer (i.e: the circle has a GestureDetector)
- The `Marker`s outside the `Circle`'s area should always be detectable.

### Do you have a potential solution?

I haven't look at it yet.

### Platforms Affected

Android, iOS, Web, Windows, MacOS, Linux

### Severity

Obtrusive: Prevents normal functioning but causes no errors in the console

### Frequency

Consistently: Always occurs at the same time and location

### Requirements

- [X] I agree to follow this project's [Code of Conduct](https://github.com/fleaflet/flutter_map/blob/master/CODE_OF_CONDUCT.md)
- [X] My Flutter/Dart installation is unaltered, and `flutter doctor` finds no relevant issues
- [X] I am using the [latest stable version](https://pub.dev/packages/flutter_map) of this package
- [X] I have checked the FAQs section on the documentation website
- [X] I have checked for similar issues which may be duplicates

**Comment by ibrierley** [comment]
Can you wrap the circle widgets in a IgnorePointer() widget ?

**Comment by daniloapr** [comment]
@ibrierley No, because they are not Widgets but `CircleMarker` object instead.

```dart
  CircleMarker _circleMarker({required LatLng center, required double radius}) {
    return CircleMarker(
      point: center,
      radius: radius,
      color: AppColors.transparentRedLight,
      borderColor: AppColors.transparentRed,
      borderStrokeWidth: 2,
      useRadiusInMeter: true,
    );
  }
```

**Comment by ibrierley** [comment]
But CircleLayer is a widget ?

**Comment by daniloapr** [comment]
You are right. It works as expected with the Ignore pointer over the `CircleLayer`.

**Comment by daniloapr** [comment]
As neither the `CircleLayer` nor `CircleMarker` has a `onPressed` callback, looks like the solution is simply adding the `IgnorePointer` on top of the `CircleLayer` build.

**Comment by JaffaKetchup** [comment]
Hey, is this still an issue?

**Comment by JaffaKetchup** [comment]
This is expected behaviour.

The children list works like a `Stack`. The last layer absorbs gestures where possible (ie. unless it is wrapped with an `IgnorePointer`. This can be reproduced without the circle layer, just by having a marker layer (with tappable markers) and tile layer in the incorrect order. It is not possible to selectively absorb/passthrough gestures inside a layer to another layer, as the layer widget fills the entire screen.

There is an issue here: this is undocumented behaviour, and this behaviour is a hot-spot for confusion. I'll add to the docs and FAQ for v6.

Closing as invalid now, as this is not technically an issue with the code. Thanks for reporting!

**Comment by JaffaKetchup** [comment]
This is re-opened for #1615 to track this. It includes a resolution for this partial bug/feature request.

**Comment by JaffaKetchup** [comment]
Closed in favour of #1616.
